src/Adapter/Product/QueryHandler/GetEditableCombinationsListHandler.php

```diff
@@ -118,10 +143,15 @@ private function formatCombinationsForEditing(
                 }
             }
 
-            $combinationsForEditing[] = new CombinationForEditing(
+            $impactOnPrice = new DecimalNumber($combination['price']);
+            $combinationsForEditing[] = new EditableCombinationForListing(
                 $combinationId,
                 $this->buildCombinationName($combinationAttributesInformation),
-                $combinationAttributesInformation
+                $combinationAttributesInformation,
+                (bool) $combination['default_on'],
+                $impactOnPrice,
+                $productPrice->plus($impactOnPrice),
+                $this->stockAvailableRepository->getForCombination(new CombinationId($combinationId))->quantity
```

Comment:
This one's nice, but we are not always sure that the `StockAvailable` entity has been created So like in product query I split into two possibilities:
- try to get the `StockAvailable` and listen to `StockAvailableNotFoundException`
- if the exception is thrown use `$combination['quantity']` as a fallback

It's not ideal but it should avoid quite some bugs, anyway the `StockAvailable` is created as soon as we udpate stock quantity so I guess it should only be absent when the product is created, and even this is not assured since we should set its quantity to zero right away 🤔

Reply:
then maybe we should get back the quantity straight from repository. (do the fallback inside)

Reply:
I don't like all this stuff with stockAvailable :smile: 
Why should we have some kind of fallback?
If we use stockAvailable then we don't use $combination->quantity or vice versa.

We should explore this and make sure there is only one reliable source.
Although it might not be so easy to achieve it :sweat:

Reply:
but why, its something wrong :thinking: 
StockAvailable should be created for each combination, shouldn't it? Do we really need a fallback? if its not created during generation of combinations it means there is already some fatal error, no?

Reply:
The thing is I don't think it is explicitely created on Combination creation, but it should
So maybe we can consider that it must be created and never catch the exception, this way of it doesn't exist it means we did not create it correctly and we will have to fix it It's probably a more stable/clean approach

Then I think we also should remove the fallback in the Product query, even if it's not exactly the scope of this PR it's a small modification (only code to be removed) and it will make the whole system consistent

Reply:
Ok so after a check, the StockAvailable is created by both Product::add and Combination::add because they both call `StockAvailable::setProductOutOfStock`

So the only thing that could go wrong is:
- for product if no shop ids are in the context (since it loops through it)
- for both it could fail when we are in a shop context which had no matching StockAvailable